src/journal/journald-server.c

```diff
@@ -1175,7 +1180,10 @@ void server_dispatch_message(
                                       NULL);
 
 finish:
-        dispatch_message_real(s, iovec, n, m, ucred, tv, label, label_len, unit_id, priority, object_pid);
+        /* restore cgroup path for logging */
+        if (c)
+                *c = '/';
+        dispatch_message_real(s, iovec, n, m, ucred, tv, label, label_len, unit_id, priority, object_pid, path);
```

Comment:
But passing NULL will use same cgroup path anyway since `cg_pid_get_path_shifted()` in `dispatcch_message_real()` will return the same path for the pid IMHO.